osu.Game.Rulesets.Osu/Edit/Blueprints/Sliders/SliderPlacementBlueprint.cs

```diff
@@ -180,7 +180,7 @@ protected override bool OnDragStart(DragStartEvent e)
                                 return false;
 
                             bSplineBuilder.Clear();
-                            bSplineBuilder.AddLinearPoint(ToLocalSpace(e.ScreenSpaceMousePosition) - HitObject.Position);
+                            bSplineBuilder.AddLinearPoint(ToLocalSpace(e.ScreenSpaceMouseDownPosition) - HitObject.Position);
```

Comment:
Unsure if this fix is accurate / sensible. Shouldn't we be reading out the position of the already-placed first control point of the slider instead of relying on a reading of the mouse position at time of drag start? Mouse position of any kind should probably just be a fallback in case there is no starting control point yet.

Reply:
I currently can't see a circumstance wherein the two positions in questions would ever be different, but maybe you've seen something that contradicts this...?

Reply:
They should not be, but them not being very much relies on OnMouseDown and OnDragStart always providing the same mouse down positions. When working on this originally, there have been somewhat significant deviations on when exactly OnDragStart fired (that's why there's all this checking logic above in the first place), so it does not feel so great to me to rely on details of input event data or timing here - especially since the intention is "start the bspline builder from where the slider started" anyway. Better to express that intention directly than in this roundabout way imo.

Reply:
I don't quite understand this train of thought.

The previous logic was using a location that is quite far away from where the user actually mouse-downs. That means that the input points to the b-spline approximator would start from a point ~10 pixels away from the actual start of the slider, potentially leading to a less-than-optimal curve (as the approximator is missing that initial data).

The only argument I could think of is that the approximator is now receiving a "jump" in data between the first point and when the drag events start to arrive and this could throw it off? But I would see that as an issue with the approximator, not the calling code.

Reply:
If OnDragStarts ScreenSpaceMouseDownPosition differs from the position that OnMouseDown received to place the first control point of the slider, the starting position of the slider will be shifted to the OnDragStart position instead of remaining where the user originally placed the slider.
After messing around with it, this even causes a very noticeable bug: Place a slider (mousedown), move your cursor, start the drag at a different position.


https://github.com/ppy/osu/assets/4215321/1ee561c4-7252-4c6c-a483-ad12dd56aabe

Reply:
That's a regression for sure, but not related to the mouse down position. I'll fix that by bringing back the "cursor point" check I was trying to get rid of.

Reply:
After investigating this behaviour a bit further, it turns out there is another issue related to directly reading out the mouse position when beginning free-hand drawing. If distance snap (or, presumably, grid snap), causes the hitobjects position to deviate from the cursor position the moment OnDragStart triggers, weird artifacting (similar to the video above) starts to happen again.

I have tried to address this in #25539